core/templates/pages/topic-editor-page/services/topic-editor-state.service.ts

```diff
@@ -97,7 +97,7 @@ export class TopicEditorStateService {
     this._topic = new Topic(
       'id', 'Topic name loading', 'Abbrev. name loading',
       'Url Fragment loading', 'Topic description loading', 'en',
-      [], [], [], 1, 1, [], 'str', '', {}, false, '', '', []
+      [], [], [], 1, 1, [], '', '', {}, false, '', '', []
```

Comment:
On looking at this more, I don't like this constructor at all. Why are we putting in all sorts of random stuff here? We should wait for the actual topic data to come to the frontend and then call the initializer (with the correct data), surely -- perhaps showing a loading interstitial in the meantime?

/cc @JeeveshGarg @srijanreddy98 -- do you have any comments on this? And @Kej-r03, sorry, I know this isn't your doing, but I want to avoid us keeping this sort of pattern throughout the codebase.

Reply:
@seanlip So are there any changes that I need to make here?

Reply:
As what i remember it is our plan, because we don't want to use interstitial more, see issue [#13637](https://github.com/oppia/oppia/issues/13637)

Reply:
Thanks Jeevesh.

@Kej-r03 Per the issue Jeevesh pointed to, I think the right thing to do is to remove these temporary objects from being constructed. Instead, the page should show "Loading..." until the data is received from the backend, and then initialize the object correctly from the get-go. Do you want to try your hand at doing this for the topic editor page? (You will need to analyze what happens on page load and understand the current flow so that you can modify it accordingly, which might be harder than what you planned to fix initially, so it's fine if you'd like to pass and take a different issue instead.)

Reply:
ok @seanlip I would like give it a shot.